facebook-common/build.gradle

```diff
@@ -60,8 +61,8 @@ android {
     }
 
     compileOptions {
-        sourceCompatibility JavaVersion.VERSION_1_7
-        targetCompatibility JavaVersion.VERSION_1_7
+        sourceCompatibility JavaVersion.VERSION_1_8
```

Comment:
Does it really necessary? I just worry that with this change it will be harder to convince FB to merge those changes, it will block updates for quite many projects.
I would rather create a separate issue and discuss possible bump to Java 8

Personally I think it's good chanage, but shouldn't be done in context of AndroidX migration (if possible)

Reply:
Note: we had a similar problem, because we were using 1.7 and we were forced to update Butterknife which started publishing a 1.8 class format jar. We solved it by:
```
    compileOptions {
        // keep 1.7 for our code, but allow D8(dex) desugaring of libraries compiled with Java 8 (Butterknife)
        sourceCompatibility JavaVersion.VERSION_1_7
        targetCompatibility JavaVersion.VERSION_1_8
    }
```
in the end the AGP tooling will dex the jar anyway, so it shouldn't be an issue on < API 24 (where Java 8 was introduced).

Reply:
It's not an issue before Android 24, desugaring is working, but some projects still not migrated yet, I just think that it's fine in general, I would just like to have this PR merged, and worry that migration to Java 8 decreases our chances to this PR be merged by FB